### ISSUE TITLE
Different content paddings and margins on page 1 and following pages

### ISSUE THREAD

**Issue by Deep-Winter** [QuestPDF/QuestPDF#167]
For letters printed on letter paper it is recommended that page one can have different content paddings/margins as the following pages. Is there a possibility to render the pdf like this?

**Comment by MarcinZiabek** [comment]
Yes 😁 You can compose your document with pages of different settings (size, margin, color, content, watermark). [Please take a lookhere](https://www.questpdf.com/documentation/patterns-and-practices.html#page-settings).

**Comment by Deep-Winter** [comment]
Thanks for the answer, but with composed pages I have more than one content area. I have one content container for the letter with length more than one page. In this case on page one for example the margin-right should be 6cm because there are some thinks printed on the letter paper and on the following pages only  2cm default margin.

**Comment by MarcinZiabek** [comment]
I think that I understand the requirement but not necessarily the use case. Would you like to describe it with a mockup/sketch?

Are you able to predict how much content is visible on the first page? Maybe you can apply additional padding to that content. It should be straightforward.

**Comment by Deep-Winter** [comment]
<img width="379" alt="Bildschirmfoto 2022-04-05 um 21 40 41" src="https://user-images.githubusercontent.com/6626971/161836534-fcb82768-481b-40e7-af56-10f983ee914d.png">
In this case for example on the first page the left margin is maybe 10cm... my content is a text with a length over 3 pages. But on page 2 and 3 I have normal page bounds for the content. So I do not need the 10cm margin.

**Comment by Deep-Winter** [comment]
Yes could be a possible solution. Page one with a big margin in the settings and a dynamic component to calculate the part of the text, which fits on the first page. And then a second page with the second part of the content in the content container. I will have a look to dynamic components, if they are released. Thank you!

**Comment by MarcinZiabek** [comment]
Or (this is only my assumption) you can create a container with the text element that preserves its state across pages. So the dynamic component only needs to apply a condition depending on the page number. No text length calculation involved 😁

However, I was implementing dynamic components quite a long time ago, so I don't remember all details involved. I just feel that I should not postpone this feature any longer even if the API may change over time. Therefore the decision of preparing experimental release.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this issue due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Since this issue was opened over 4 years ago, the library has undergone significant changes. Elements like `ShowOnce`, `SkipOnce`, and `ShowIf` now allow conditional rendering based on page context, which can be used to achieve different layouts on the first page vs subsequent pages.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
